### PR TITLE
Timothy mcgowen

### PR DESCRIPTION
**Pull request by mcgowent**

### CODE REVIEW

File: client/src/components/PrivateRoute.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
Comment (AdetunjiShennaike) [inline client/src/components/PrivateRoute.js:1]: nice nice mr fancy

File: client/src/App.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React, { useState } from "react";
-import { BrowserRouter as Router, Route } from "react-router-dom";
+import { BrowserRouter as Router, Route, Link } from "react-router-dom";
```
Comment (AdetunjiShennaike) [inline client/src/App.js:2]: add some styling with navLink

File: server.js [hunk 3]
```diff
@@ -102,7 +102,7 @@ function authenticator(req, res, next) {
 
 app.post("/api/login", (req, res) => {
   const { username, password } = req.body;
-  if (username === "Lambda School" && password === "i<3Lambd4") {
+  if (username === "Tim" && password === "password") {
```
Comment (AdetunjiShennaike) [inline server.js:105]: hahaah the change

File: client/src/components/BubblePage.js [hunk 4]
```diff
@@ -10,15 +10,21 @@ const BubblePage = () => {
   // fetch your colors data from the server when the component mounts
```
Comment (AdetunjiShennaike) [inline client/src/components/BubblePage.js:10]: addd comments!!!!

File: client/src/components/ColorList.js [hunk 5]
```diff
@@ -19,23 +21,33 @@ const ColorList = ({ colors, updateColors, history }) => {
...
-    // Make a put request to save your updated color
-    // think about where will you get the id from...
-    // where is is saved right now? props
-    console.log('THIS IS INSIDE saveEDIT', colorToEdit);
     axiosWithAuth()
-      .put(`http://localhost:5000/api/colors/:id${colorToEdit.id}`, colorToEdit)
+      .put(`http://localhost:5000/api/colors/:${colorToEdit.id}`, colorToEdit)
       .then(res => {
         console.log('Success!', res);
+        let newColors = colors.map(e => {
```
Comment (AdetunjiShennaike) [inline client/src/components/ColorList.js:28]: nice logic ;)

File: client/src/components/ColorList.js [hunk 6]
```diff
@@ -19,23 +21,33 @@ const ColorList = ({ colors, updateColors, history }) => {
...
   };
 
   const deleteColor = color => {
     // make a delete request to delete this color
+    console.log('This should be a color ID', color.id)
+    axiosWithAuth()
+      .delete(`http://localhost:5000/api/colors/:${color.id}`)
+      .then(res => {
+        console.log('Deleted!', res);
+        let newColors = colors.filter(e => e.id !== color.id)
```
Comment (AdetunjiShennaike) [inline client/src/components/ColorList.js:47]: good filter i like it

File: README.md [hunk 7]
```diff
@@ -29,10 +29,20 @@ In this project you will create a login page and request a token from the server
...
+
+  Use tokens and not actual info being send and saved around the site
+
 - [ ] Describe how web servers work.
+
+  It is a computer that recieves and sends back data
+
 - [ ] Which HTTP methods can be mapped to the CRUD acronym that we use when interfacing with APIs/Servers.
 
+  Update and Delete
```
Comment (AdetunjiShennaike) [inline README.md:45]: Create - Post
Read - Get
Update - Put 
Delete - Delete

File: README.md [hunk 8]
```diff
@@ -29,10 +29,20 @@ In this project you will create a login page and request a token from the server
 Demonstrate your understanding of this Sprint's concepts by answering the following free-form questions. Edit this document to include your answers after each question. Make sure to leave a blank line above and below your answer so it is clear and easy to read by your project manager.
 
 - [ ] Explain what a token is used for.
+
```
Comment (AdetunjiShennaike) [inline README.md:32]: short answers =]

**Review by AdetunjiShennaike — APPROVED** [review]

File: client/src/components/BubblePage.js (hunk 4 above)
Comment (mcgowent) [reply client/src/components/BubblePage.js:10]: Nope

**Pull request merged**